### PR TITLE
Pension 90725 added rescue blocks for confirmation email send attempts

### PR DESCRIPTION
**Pull request by danlim715**
## Summary

- *This work is behind a feature toggle (flipper): NO*
- Added rescue blocks for send_confirmation_email attempts after claim submission. This should be a caught exception so submission job doesn't fail and attempt retries.
- Pension benefits team to own this change.

## Related issue(s)

- [*Link to ticket created in va.gov-team repo*](https://github.com/department-of-veterans-affairs/va.gov-team/issues/90725)

## Testing done

- [x] *New code is covered by unit tests*
- Some attempts to send confirmation emails would fail and cause the claim submission job to fail and trigger a retry.
- Now if a send confirmation email attempt fails, it'll log the failed attempt to console and Datadog, and not trigger a claim submission retry.

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)

### CODE REVIEW

**Review by balexandr — APPROVED** [review]
LGTM

File: modules/pensions/app/sidekiq/pensions/pension_benefit_intake_job.rb [hunk 1]
```diff
@@ -59,7 +59,12 @@ def perform(saved_claim_id, user_account_uuid = nil)
 
       upload_document
 
-      @claim.send_confirmation_email if @claim.respond_to?(:send_confirmation_email)
+      begin
+        @claim.send_confirmation_email if @claim.respond_to?(:send_confirmation_email)
+      rescue e
+        @pension_monitor.track_send_confirmation_email_failure(@claim, @intake_service, @user_account_uuid, e)
+      end
```
Comment (wayne-weibel) [inline modules/pensions/app/sidekiq/pensions/pension_benefit_intake_job.rb:66]: need to add to unit tests to cover the rescue block

**Review by wayne-weibel — CHANGES_REQUESTED** [review]
test coverage change
failing test will be resolved once merge conflict is resolved

File: modules/pensions/app/sidekiq/pensions/pension_benefit_intake_job.rb (hunk 1 above)
Comment (danlim715) [reply modules/pensions/app/sidekiq/pensions/pension_benefit_intake_job.rb:66]: @wayne-weibel i'll need assistance with this, as I will be on PTO.

**Review by wayne-weibel — APPROVED** [review]

**Review by ryan-mcneil — APPROVED** [review]

**Pull request merged**